# Tinder: speed dating data analysis

## Goal
This project aims at understanding **what drives mutual interest during speed dating** and which features influence a participant’s decision to agree to a second date.

## Dataset
Data was collected from participants in experimental speed dating events organized by Tinder and held between 2002 and 2004. Each participant had a four minute "first date" with every participant of the opposite sex. After each date, participants indicated whether they would like a second date and rated their date on six attributes: attractiveness, sincerity, intelligence, fun, ambition, and shared interests.

The dataset also includes questionnaire data collected at different points in the process, covering:
- Demographics
- Lifestyle information
- Dating habits
- Self-perception across key attributes
- Beliefs about what others find valuable in a partner

[Dataset](https://full-stack-assets.s3.eu-west-3.amazonaws.com/M03-EDA/Speed+Dating+Data.csv)

[Dataset Description](https://full-stack-assets.s3.eu-west-3.amazonaws.com/M03-EDA/Speed+Dating+Data+Key.doc)

## Methodology
This explanatory data analysis follows these steps:
- initial exploration and overview of the dataset;
- data cleaning; 
- general descriptive statistics;
- detailed descriptive analysis and visualisations about:
    - analysis of the valuable attributes in a partner, based on pre-experiment responses
    - comparison of actual behavior with stated preferences
    - assessment of participants' ability to gauge their own perceived value in the dating market
    - influence of the round order on the decision to request a second date

In [1]:
# Import packages
import numpy as np
import pandas as pd
import plotly.express as px
import plotly.io as pio
import plotly.graph_objects as go
from plotly.subplots import make_subplots

pd.options.mode.chained_assignment = None  # default='warn'

# Option which displays all columns and rows of dataframes
pd.options.display.max_columns = None
pd.options.display.max_rows = None

In [2]:
# Load data
filename = "data/Speed+Dating+Data.csv"
#filename = "https://full-stack-assets.s3.eu-west-3.amazonaws.com/M03-EDA/Speed+Dating+Data.csv"

# Using the default encoding UTF-8 does not work (UnicodeDecodeError), instead use Using Latin-1 encoding (ISO-8859-1).
df = pd.read_csv(filename, 
                encoding = "ISO-8859-1")

# 1. Initial exploration

In [3]:
# Display first 5 rows
df.head()

,iid,id,gender,idg,condtn,wave,round,position,positin1,order,partner,pid,match,int_corr,samerace,age_o,race_o,pf_o_att,pf_o_sin,pf_o_int,pf_o_fun,pf_o_amb,pf_o_sha,dec_o,attr_o,sinc_o,intel_o,fun_o,amb_o,shar_o,like_o,prob_o,met_o,age,field,field_cd,undergra,mn_sat,tuition,race,imprace,imprelig,from,zipcode,income,goal,date,go_out,career,career_c,sports,tvsports,exercise,dining,museums,art,hiking,gaming,clubbing,reading,tv,theater,movies,concerts,music,shopping,yoga,exphappy,expnum,attr1_1,sinc1_1,intel1_1,fun1_1,amb1_1,shar1_1,attr4_1,sinc4_1,intel4_1,fun4_1,amb4_1,shar4_1,attr2_1,sinc2_1,intel2_1,fun2_1,amb2_1,shar2_1,attr3_1,sinc3_1,fun3_1,intel3_1,amb3_1,attr5_1,sinc5_1,intel5_1,fun5_1,amb5_1,dec,attr,sinc,intel,fun,amb,shar,like,prob,met,match_es,attr1_s,sinc1_s,intel1_s,fun1_s,amb1_s,shar1_s,attr3_s,sinc3_s,intel3_s,fun3_s,amb3_s,satis_2,length,numdat_2,attr7_2,sinc7_2,intel7_2,fun7_2,amb7_2,shar7_2,attr1_2,sinc1_2,intel1_2,fun1_2,amb1_2,shar1_2,attr4_2,sinc4_2,intel4_2,fun4_2,amb4_2,shar4_2,attr2_2,sinc2_2,intel2_2,fun2_2,amb2_2,shar2_2,attr3_2,sinc3_2,intel3_2,fun3_2,amb3_2,attr5_2,sinc5_2,intel5_2,fun5_2,amb5_2,you_call,them_cal,date_3,numdat_3,num_in_3,attr1_3,sinc1_3,intel1_3,fun1_3,amb1_3,shar1_3,attr7_3,sinc7_3,intel7_3,fun7_3,amb7_3,shar7_3,attr4_3,sinc4_3,intel4_3,fun4_3,amb4_3,shar4_3,attr2_3,sinc2_3,intel2_3,fun2_3,amb2_3,shar2_3,attr3_3,sinc3_3,intel3_3,fun3_3,amb3_3,attr5_3,sinc5_3,intel5_3,fun5_3,amb5_3
0,1,1.0,0,1,1,1,10,7,NaN,4,1,11.0,0,0.14,0,27.0,2.0,35.0,20.0,20.0,20.0,0.0,5.0,0,6.0,8.0,8.0,8.0,8.0,6.0,7.0,4.0,2.0,21.0,Law,1.0,NaN,NaN,NaN,4.0,2.0,4.0,Chicago,"60,521","69,487.00",2.0,7.0,1.0,lawyer,NaN,9.0,2.0,8.0,9.0,1.0,1.0,5.0,1.0,5.0,6.0,9.0,1.0,10.0,10.0,9.0,8.0,1.0,3.0,2.0,15.0,20.0,20.0,15.0,15.0,15.0,NaN,NaN,NaN,NaN,NaN,NaN,35.0,20.0,15.0,20.0,5.0,5.0,6.0,8.0,8.0,8.0,7.0,NaN,NaN,NaN,NaN,NaN,1,6.0,9.0,7.0,7.0,6.0,5.0,7.0,6.0,2.0,4.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,6.0,2.0,1.0,NaN,NaN,NaN,NaN,NaN,NaN,19.44,16.67,13.89,22.22,11.11,16.67,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,6.0,7.0,8.0,7.0,6.0,NaN,NaN,NaN,NaN,NaN,1.0,1.0,0.0,NaN,NaN,15.0,20.0,20.0,15.0,15.0,15.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,5.0,7.0,7.0,7.0,7.0,NaN,NaN,NaN,NaN,NaN
1,1,1.0,0,1,1,1,10,7,NaN,3,2,12.0,0,0.54,0,22.0,2.0,60.0,0.0,0.0,40.0,0.0,0.0,0,7.0,8.0,10.0,7.0,7.0,5.0,8.0,4.0,2.0,21.0,Law,1.0,NaN,NaN,NaN,4.0,2.0,4.0,Chicago,"60,521","69,487.00",2.0,7.0,1.0,lawyer,NaN,9.0,2.0,8.0,9.0,1.0,1.0,5.0,1.0,5.0,6.0,9.0,1.0,10.0,10.0,9.0,8.0,1.0,3.0,2.0,15.0,20.0,20.0,15.0,15.0,15.0,NaN,NaN,NaN,NaN,NaN,NaN,35.0,20.0,15.0,20.0,5.0,5.0,6.0,8.0,8.0,8.0,7.0,NaN,NaN,NaN,NaN,NaN,1,7.0,8.0,7.0,8.0,5.0,6.0,7.0,5.0,1.0,4.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,6.0,2.0,1.0,NaN,NaN,NaN,NaN,NaN,NaN,19.44,16.67,13.89,22.22,11.11,16.67,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,6.0,7.0,8.0,7.0,6.0,NaN,NaN,NaN,NaN,NaN,1.0,1.0,0.0,NaN,NaN,15.0,20.0,20.0,15.0,15.0,15.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,5.0,7.0,7.0,7.0,7.0,NaN,NaN,NaN,NaN,NaN
2,1,1.0,0,1,1,1,10,7,NaN,10,3,13.0,1,0.16,1,22.0,4.0,19.0,18.0,19.0,18.0,14.0,12.0,1,10.0,10.0,10.0,10.0,10.0,10.0,10.0,10.0,1.0,21.0,Law,1.0,NaN,NaN,NaN,4.0,2.0,4.0,Chicago,"60,521","69,487.00",2.0,7.0,1.0,lawyer,NaN,9.0,2.0,8.0,9.0,1.0,1.0,5.0,1.0,5.0,6.0,9.0,1.0,10.0,10.0,9.0,8.0,1.0,3.0,2.0,15.0,20.0,20.0,15.0,15.0,15.0,NaN,NaN,NaN,NaN,NaN,NaN,35.0,20.0,15.0,20.0,5.0,5.0,6.0,8.0,8.0,8.0,7.0,NaN,NaN,NaN,NaN,NaN,1,5.0,8.0,9.0,8.0,5.0,7.0,7.0,NaN,1.0,4.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,6.0,2.0,1.0,NaN,NaN,NaN,NaN,NaN,NaN,19.44,16.67,13.89,22.22,11.11,16.67,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,6.0,7.0,8.0,7.0,6.0,NaN,NaN,NaN,NaN,NaN,1.0,1.0,0.0,NaN,NaN,15.0,20.0,20.0,15.0,15.0,15.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,5.0,7.0,7.0,7.0,7.0,NaN,NaN,NaN,NaN,NaN
3,1,1.0,0,1,1,1,10,7,NaN,5,4,14.0,1,0.61,0,23.0,2.0,30.0,5.0,15.0,40.0,5.0,5.0,1,7.0,8.0,9.0,8.0,9

In [4]:
# Display the shape of the dataframe
df.shape

(8378, 195)

In [5]:
# Display the variables types
df.dtypes

iid           int64
id          float64
gender        int64
idg           int64
condtn        int64
wave          int64
round         int64
position      int64
positin1    float64
order         int64
partner       int64
pid         float64
match         int64
int_corr    float64
samerace      int64
age_o       float64
race_o      float64
pf_o_att    float64
pf_o_sin    float64
pf_o_int    float64
pf_o_fun    float64
pf_o_amb    float64
pf_o_sha    float64
dec_o         int64
attr_o      float64
sinc_o      float64
intel_o     float64
fun_o       float64
amb_o       float64
shar_o      float64
like_o      float64
prob_o      float64
met_o       float64
age         float64
field        object
field_cd    float64
undergra     object
mn_sat       object
tuition      object
race        float64
imprace     float64
imprelig    float64
from         object
zipcode      object
income       object
goal        float64
date        float64
go_out      float64
career       object
career_c    float64


In [6]:
# Display descriptive statistics
df.describe(include = "all").round(2)

,iid,id,gender,idg,condtn,wave,round,position,positin1,order,partner,pid,match,int_corr,samerace,age_o,race_o,pf_o_att,pf_o_sin,pf_o_int,pf_o_fun,pf_o_amb,pf_o_sha,dec_o,attr_o,sinc_o,intel_o,fun_o,amb_o,shar_o,like_o,prob_o,met_o,age,field,field_cd,undergra,mn_sat,tuition,race,imprace,imprelig,from,zipcode,income,goal,date,go_out,career,career_c,sports,tvsports,exercise,dining,museums,art,hiking,gaming,clubbing,reading,tv,theater,movies,concerts,music,shopping,yoga,exphappy,expnum,attr1_1,sinc1_1,intel1_1,fun1_1,amb1_1,shar1_1,attr4_1,sinc4_1,intel4_1,fun4_1,amb4_1,shar4_1,attr2_1,sinc2_1,intel2_1,fun2_1,amb2_1,shar2_1,attr3_1,sinc3_1,fun3_1,intel3_1,amb3_1,attr5_1,sinc5_1,intel5_1,fun5_1,amb5_1,dec,attr,sinc,intel,fun,amb,shar,like,prob,met,match_es,attr1_s,sinc1_s,intel1_s,fun1_s,amb1_s,shar1_s,attr3_s,sinc3_s,intel3_s,fun3_s,amb3_s,satis_2,length,numdat_2,attr7_2,sinc7_2,intel7_2,fun7_2,amb7_2,shar7_2,attr1_2,sinc1_2,intel1_2,fun1_2,amb1_2,shar1_2,attr4_2,sinc4_2,intel4_2,fun4_2,amb4_2,shar4_2,attr2_2,sinc2_2,intel2_2,fun2_2,amb2_2,shar2_2,attr3_2,sinc3_2,intel3_2,fun3_2,amb3_2,attr5_2,sinc5_2,intel5_2,fun5_2,amb5_2,you_call,them_cal,date_3,numdat_3,num_in_3,attr1_3,sinc1_3,intel1_3,fun1_3,amb1_3,shar1_3,attr7_3,sinc7_3,intel7_3,fun7_3,amb7_3,shar7_3,attr4_3,sinc4_3,intel4_3,fun4_3,amb4_3,shar4_3,attr2_3,sinc2_3,intel2_3,fun2_3,amb2_3,shar2_3,attr3_3,sinc3_3,intel3_3,fun3_3,amb3_3,attr5_3,sinc5_3,intel5_3,fun5_3,amb5_3
count,8378.00,8377.00,8378.0,8378.00,8378.00,8378.00,8378.00,8378.00,6532.00,8378.00,8378.00,8368.00,8378.00,8220.00,8378.00,8274.00,8305.00,8289.00,8289.00,8289.00,8280.00,8271.00,8249.00,8378.00,8166.00,8091.00,8072.00,8018.00,7656.00,7302.00,8128.00,8060.00,7993.00,8283.00,8315,8296.00,4914,3133,3583,8315.00,8299.00,8299.00,8299,7314,4279,8299.00,8281.00,8299.00,8289,8240.00,8299.00,8299.00,8299.00,8299.00,8299.00,8299.00,8299.00,8299.00,8299.00,8299.00,8299.00,8299.00,8299.00,8299.00,8299.00,8299.00,8299.00,8277.00,1800.00,8299.00,8299.00,8299.00,8289.00,8279.00,8257.00,6489.00,6489.00,6489.00,6489.00,6489.00,6467.00,8299.00,8299.00,8299.00,8299.00,8289.00,8289.00,8273.00,8273.00,8273.00,8273.00,8273.00,4906.00,4906.00,4906.00,4906.00,4906.00,8378.00,8176.00,8101.00,8082.00,8028.00,7666.00,7311.00,8138.00,8069.00,8003.00,7205.00,4096.00,4096.00,4096.00,4096.00,4096.00,4096.00,4000.00,4000.00,4000.00,4000.00,4000.00,7463.00,7463.00,7433.00,1984.00,1955.00,1984.00,1984.00,1955.00,1974.00,7445.00,7463.00,7463.00,7463.00,7463.00,7463.00,5775.00,5775.00,5775.00,5775.00,5775.00,5775.00,5775.00,5775.00,5775.00,5775.00,5775.00,5775.00,7463.00,7463.00,7463.00,7463.00,7463.00,4377.00,4377.00,4377.00,4377.00,4377.00,3974.00,3974.00,3974.00,1496.00,668.00,3974.00,3974.00,3974.00,3974.00,3974.0,3974.00,2016.00,2016.00,2016.00,2016.00,2016.00,2016.00,2959.00,2959.00,2959.00,2959.00,2959.00,2959.00,2959.00,2959.00,2959.00,2959.00,2959.00,2016.00,3974.00,3974.00,3974.00,3974.00,3974.00,2016.00,2016.00,2016.00,2016.00,2016.00
unique,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,259,NaN,241,68,115,NaN,NaN,NaN,269,409,261,NaN,NaN,NaN,367,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
top,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Business,NaN,UC Berkeley,"1,400.00","26,908.00",NaN,NaN,NaN,New York,0,"55,080.00",NaN,N

# 2. Data cleaning

This section is dedicated to:
- Dealing with missing values: checking and addressing missing values
- Performing data validation: updating data types, checking ranges for numerical data.

In [7]:
# Display the number of missing values
df_count_na = df.isnull().sum().reset_index().rename(columns = {'index' : 'var_name', 0: "count_of_missing_values"})
df_count_na['proportion_of_missing_values'] = round(df_count_na['count_of_missing_values'] / len(df) * 100, 2)
df_count_na.sort_values(['proportion_of_missing_values'], 
                        ascending = False,
                        inplace = True)
df_count_na

,var_name,count_of_missing_values,proportion_of_missing_values
160,num_in_3,7710,92.03
159,numdat_3,6882,82.14
68,expnum,6578,78.52
123,sinc7_2,6423,76.67
126,amb7_2,6423,76.67
127,shar7_2,6404,76.44
122,attr7_2,6394,76.32
125,fun7_2,6394,76.32
124,intel7_2,6394,76.32
193,fun5_3,6362,75.94


In [8]:
# Focus on variables which have a significant proportion of missing values (above 30%)

count_na_prct30 = len(df_count_na[df_count_na['proportion_of_missing_values'] >= 30])

print(f"{count_na_prct30} variables \
(accounting for {round(count_na_prct30/ len(df_count_na),2)*100} % of the total number of variables) \
have above 30% of missing observations")

# Visual representation 

fig = px.bar(df_count_na[df_count_na['proportion_of_missing_values'] >= 30], 
            x = 'var_name', 
            y = 'proportion_of_missing_values',
            title = "Percentage of missing values by variable (for variables with more than 30% of missing values)")
fig.update_layout(
    xaxis_title = 'Variable',
    yaxis_title = 'Percentage of missing values (%)',
    xaxis_tickangle = -45
)
fig.show()

83 variables (accounting for 43.0 % of the total number of variables) have above 30% of missing observations


Variables with a high percentage of missing values are often those related to the survey after the date experiment.

We decide to drop the variables with more than 70% of missing values: we are sure that we will not use them in the analysis.

In [9]:
# Removing variables with more than 70% of missing values
cols_to_remove = df_count_na[df_count_na['proportion_of_missing_values'] >= 70]['var_name'].tolist()
print(f"{len(cols_to_remove)} columns with more than 70% missing values will be removed from the dataset.")
df = df.drop(columns = cols_to_remove)
print(df.shape)

21 columns with more than 70% missing values will be removed from the dataset.
(8378, 174)


In [10]:
# Filling id missing value which can be inferred
df.loc[(df['iid'] == 552) & (df['id'].isnull()), 'id'] = 22

# Convert id to integer
df['iid'] = df['iid'].astype(int)

In [11]:
# Dealing with categorical variables

# Converting gender, race, career field, and date frequency into categorical data type (which reduces memory footprint 
# and allows faster computation)

# Gender variable ("Male" or "Female")
df['gender'] = df['gender'].map({0: 'Female', 1: 'Male'})
df['gender'].value_counts().reset_index()

,gender,count
0,Male,4194
1,Female,4184


In [12]:
# Race
df['race'] = df['race'].map({1: 'Black/African American', 
                            2: 'European/Caucasian American',
                            3: 'Latino/Hispanic American',
                            4: 'Asian/Pacific Islander American',
                            5: 'Native American',
                            6: 'Other'})\
                        .astype('category')

df['race'].value_counts(dropna = False).reset_index()

,race,count
0,European/Caucasian American,4727
1,Asian/Pacific Islander American,1982
2,Latino/Hispanic American,664
3,Other,522
4,Black/African American,420
5,NaN,63


In [13]:
# Goals
goal_mapping = {
    1: 'Have fun',
    2: 'Meet new people',
    3: 'Get a date',
    4: 'Serious relationship',
    5: 'I did it',
    6: 'Other'
}

df['goal'] = df['goal'].map(goal_mapping).astype('category')
df['goal'].value_counts(dropna = False).reset_index()

,goal,count
0,Have fun,3426
1,Meet new people,3012
2,Get a date,631
3,I did it,510
4,Other,419
5,Serious relationship,301
6,NaN,79


In [14]:
# Date and go out frequency
freq_mapping = {
    1: 'Several times a week', 
    2: 'Twice a week',
    3: 'Once a week',
    4: 'Twice a month',
    5: 'Once a month',
    6: 'Several times a year',
    7: 'Almost never'}

for var in ['date', 'go_out']:
    df[var] = df[var].map(freq_mapping).astype('category')
    display(df[var].value_counts(dropna = False).reset_index())

,date,count
0,Several times a year,2094
1,Twice a month,2040
2,Once a month,1528
3,Almost never,1434
4,Once a week,783
5,Twice a week,308
6,NaN,97
7,Several times a week,94


,go_out,count
0,Twice a week,2990
1,Several times a week,2610
2,Once a week,1949
3,Twice a month,450
4,Once a month,164
5,Several times a year,99
6,NaN,79
7,Almost never,37


# 3. Descriptive statistics

The dataset variables can be divided into two types:
- **individual variables** which describe each participant: id, socio-demographic information, beliefs and expectations about dating prior to the experiment
- **variables related to the dating experiment**: the definition of the waves, the opinion about the people the participants met during the date, the issue of the date.

We first have a look at some characteristics of the participants of the experiment.
Since the number of rounds in each wave is different, we need to work with a data subset containing observations without duplicates, so that each participant is equally treated.

In [15]:
# Subset the dataset and drop duplicates

var_subset = ['iid', 'wave', 'gender', 'age', 'field', 'field_cd', 'undergra',
        'mn_sat', 'tuition', 'race', 'imprace', 'imprelig', 'from', 'zipcode',
        'income', 'goal', 'date', 'go_out', 'career',
        'sports', 'tvsports', 'exercise', 'dining', 'museums', 'art', 'hiking', 
        'gaming', 'clubbing', 'reading', 'tv',
        'theater', 'movies', 'concerts', 'music', 'shopping', 'yoga',
        'exphappy', 
        #'expnum', 
        'attr1_1', 'sinc1_1', 'intel1_1', 'fun1_1', 'amb1_1', 'shar1_1',
        #'attr4_1', 'sinc4_1', 'intel4_1', 'fun4_1', 'amb4_1', 'shar4_1',
        #'attr2_1', 'sinc2_1', 'intel2_1', 'fun2_1', 'amb2_1', 'shar2_1',
        'attr3_1', 'sinc3_1', 'fun3_1', 'intel3_1', 'amb3_1', 
        #'attr5_1', 'sinc5_1', 'fun5_1', 'intel5_1', 'amb5_1'
        ]

df_participant = df[var_subset].drop_duplicates(var_subset)
df_participant.shape

(551, 48)

In [16]:
df_participant.info()

<class 'pandas.core.frame.DataFrame'>
Index: 551 entries, 0 to 8356
Data columns (total 48 columns):
 #   Column    Non-Null Count  Dtype   
---  ------    --------------  -----   
 0   iid       551 non-null    int64   
 1   wave      551 non-null    int64   
 2   gender    551 non-null    object  
 3   age       543 non-null    float64 
 4   field     545 non-null    object  
 5   field_cd  544 non-null    float64 
 6   undergra  315 non-null    object  
 7   mn_sat    202 non-null    object  
 8   tuition   234 non-null    object  
 9   race      545 non-null    category
 10  imprace   544 non-null    float64 
 11  imprelig  544 non-null    float64 
 12  from      544 non-null    object  
 13  zipcode   475 non-null    object  
 14  income    281 non-null    object  
 15  goal      544 non-null    category
 16  date      543 non-null    category
 17  go_out    544 non-null    category
 18  career    543 non-null    object  
 19  sports    544 non-null    float64 
 20  tvsports  544 

## 3.1. Number of participants

In [17]:
df_count_gender = df_participant['gender'].value_counts(dropna = False).reset_index()
df_count_gender['proportion (%)'] = round(100 * df_count_gender['count'] / df_count_gender['count'].sum(),1)
display(df_count_gender)

fig = px.pie(df_count_gender, 
            values = 'count', 
            names = 'gender', 
            color = 'gender',
            title = 'Repartition of participants by gender',
            color_discrete_map = {'Male' : 'steelblue',
                                'Female' : 'lightcoral'})
fig.update_layout(title_x=0.5)
fig.show()


,gender,count,proportion (%)
0,Male,277,50.3
1,Female,274,49.7


551 participants were involved in this dating experiment, with almost an equal repartition by gender.

## 3.2. Age

In [18]:
# Descriptive statistics of age
df_age = df_participant['age'].describe().round(2).reset_index()
df_age_gender = df_participant.groupby('gender', dropna = False)['age'].describe().round(2).T.reset_index()
for var in ['Female', 'Male']:
    df_age[var + '_age'] = df_age_gender[var]
df_age

,index,age,Female_age,Male_age
0,count,543.00,269.00,274.00
1,mean,26.36,26.13,26.59
2,std,3.77,4.00,3.52
3,min,18.00,19.00,18.00
4,25%,24.00,23.00,24.00
5,50%,26.00,26.00,27.00
6,75%,28.00,28.00,29.00
7,max,55.00,55.00,42.00


In [19]:
fig = px.histogram(df_participant, 
                x = "age", 
                color_discrete_sequence = ['lightseagreen'],
                marginal = "box",
                title = "Age distribution")
fig.update_layout(title_x=0.5)
fig.show()

fig = px.histogram(df_participant, 
                x = "age", 
                color = "gender",
                color_discrete_map = {
                    'Male': 'steelblue',
                    'Female': 'lightcoral'
                },
                marginal = "box",
                title = "Age distribution by gender")
fig.update_layout(title_x=0.5)
fig.show()

Most of the participants are in their twenties: half of the participants are between 24 and 28 years old.
Men are on average a little bit older than women.
We note one outlier (55 years old female).

## 3.2. Race

In [20]:
df_race = df_participant['race'].value_counts(dropna = False).reset_index()
df_race['proportion (%)'] = round(100 * df_race['count'] / df_race['count'].sum(),1)
display(df_race)

,race,count,proportion (%)
0,European/Caucasian American,304,55.2
1,Asian/Pacific Islander American,136,24.7
2,Latino/Hispanic American,42,7.6
3,Other,37,6.7
4,Black/African American,26,4.7
5,NaN,6,1.1


In [21]:
df_race_gender = df_participant.groupby(['gender'], dropna = False)['race'].value_counts(dropna = False).reset_index()

df_race_gender = df_participant.groupby(['gender'], dropna = False)['race'].value_counts(dropna = False).reset_index()

df_race_female = df_race_gender.loc[df_race_gender['gender'] == 'Female']
df_race_female['female proportion (%)'] = round(100 * df_race_female['count'] / df_race_female['count'].sum(), 1)
df_race_female = df_race_female.rename(columns = {'count' : 'female count'})

df_race_male = df_race_gender.loc[df_race_gender['gender'] == 'Male']
df_race_male['male proportion (%)'] = round(100 * df_race_male['count'] / df_race_male['count'].sum(), 1)
df_race_male = df_race_male.rename(columns = {'count' : 'male count'})

display(df_race_female)
display(df_race_male)

,gender,race,female count,female proportion (%)
0,Female,European/Caucasian American,142,51.8
1,Female,Asian/Pacific Islander American,71,25.9
2,Female,Latino/Hispanic American,25,9.1
3,Female,Black/African American,16,5.8
4,Female,Other,16,5.8
5,Female,NaN,4,1.5


,gender,race,male count,male proportion (%)
6,Male,European/Caucasian American,162,58.5
7,Male,Asian/Pacific Islander American,65,23.5
8,Male,Other,21,7.6
9,Male,Latino/Hispanic American,17,6.1
10,Male,Black/African American,10,3.6
11,Male,NaN,2,0.7


Participants are mostly European/Caucasian or Asian/Pacific Islander: these two ethnic groups represent both 80% of the number of the participants. European/Caucasian people are predominant, with more than half of the participants.


## 3.3. Goals

In [22]:
df_goal = df_participant['goal'].value_counts(dropna = False).reset_index()
df_goal['proportion (%)'] = round(100 * df_goal['count'] / df_goal['count'].sum(), 1)
display(df_goal)

,goal,count,proportion (%)
0,Have fun,228,41.4
1,Meet new people,189,34.3
2,Get a date,40,7.3
3,I did it,35,6.4
4,Other,30,5.4
5,Serious relationship,22,4.0
6,NaN,7,1.3


In [23]:
# Goals by genre
df_goal_gender = df_participant.groupby(['gender', 'goal'], observed=True).size().reset_index(name='count')
df_goal_gender['proportion (%)'] = round(100 * df_goal_gender['count'] / df_goal_gender.groupby('gender')['count'].transform('sum'), 1)
df_goal_gender

,gender,goal,count,proportion (%)
0,Female,Get a date,13,4.8
1,Female,Have fun,115,42.8
2,Female,I did it,18,6.7
3,Female,Meet new people,100,37.2
4,Female,Other,14,5.2
5,Female,Serious relationship,9,3.3
6,Male,Get a date,27,9.8
7,Male,Have fun,113,41.1
8,Male,I did it,17,6.2
9,Male,Meet new people,89,32.4


In [24]:
df_goal_gender = df_goal_gender.sort_values('count', ascending=False)

fig = px.bar(
    df_goal_gender,
    x='goal',
    y='proportion (%)',
    color='gender',
    color_discrete_map = {
        'Male': 'steelblue',
        'Female': 'lightcoral'
    },
    barmode='group',
    text='proportion (%)'
)

fig.update_layout(
    title="Distribution of goals by gender",
    xaxis_title="Goal",
    yaxis_title="Proportion (%)"
)
fig.update_layout(title_x=0.5)
fig.show()


Overall, participants are largely interested in having fun and meeting other people: these two goals represent 75% of the answers. Only 4% of the participants are looking for a serious relationship.

A higher proportion of men (10%) expect a date compared with women (5%).

## 3.4. Matching results

In [25]:
# Here we work with the full dataframe. 

# Create a variable "result" 
df['result'] = "Unknown"
df.loc[df['match'] == 1, 'result'] = "Match"
df.loc[((df['dec'] == 1) & (df['dec_o'] == 0)) | ((df['dec'] == 0) & (df['dec_o'] == 1)), 'result'] = "Frustration"
df.loc[(df['dec'] == 0) & (df['dec_o'] == 0), 'result'] = "Mutual rejection"

df_result = df['result'].value_counts(dropna = False).reset_index()
display(df_result)

fig = px.pie(
    df_result,
    values = 'count',
    names = 'result',
    color = 'result',
    title ='Matching results for overall participants',
    color_discrete_map = {'Match': 'mediumseagreen',
                        'Frustration' : 'indianred',
                        'Mutual rejection' : 'lightgray'}
)
fig.update_layout(title_x=0.5)
fig.show()

,result,count
0,Frustration,4273
1,Mutual rejection,2725
2,Match,1380


In [26]:
df_matching_female = df.loc[df['gender'] == "Female"]['result']\
    .value_counts(dropna=False)\
    .reset_index()
df_matching_female['proportion'] = round(100 * df_matching_female['count'] / df_matching_female['count'].sum(), 3)

df_matching_male = df.loc[df['gender'] == "Male"]['result']\
    .value_counts(dropna=False)\
    .reset_index()
df_matching_male['proportion'] = round(100 * df_matching_male['count'] / df_matching_male['count'].sum(), 3)

print("Matching after a speed date, for female")
display(df_matching_female)

print("Matching after a speed date, for male")
display(df_matching_male)

Matching after a speed date, for female


,result,count,proportion
0,Frustration,2135,51.028
1,Mutual rejection,1359,32.481
2,Match,690,16.491


Matching after a speed date, for male


,result,count,proportion
0,Frustration,2138,50.978
1,Mutual rejection,1366,32.570
2,Match,690,16.452


In [27]:
fig = make_subplots(
    rows=1, cols=2, 
    specs=[[{'type':'domain'}, {'type':'domain'}]],  # important pour pie chart
    subplot_titles=["Female", "Male"]
)

fig.add_trace(go.Pie(labels=df_matching_female['result'],
                    values=df_matching_female['proportion'],
                    textinfo='label+percent',
                    marker=dict(colors=['indianred', 'lightgray', 'mediumseagreen'])), 
            row=1, col=1)

fig.add_trace(go.Pie(labels=df_matching_male['result'],
                    values=df_matching_male['proportion'],
                    textinfo='label+percent',
                    marker=dict(colors=['indianred', 'lightgray', 'mediumseagreen'])),
            row=1, col=2)

fig.update_layout(title_text="Matching distribution by genre", title_x=0.5)
fig.show()

In [28]:
# Decision per genre

decision_mapping = {1: 'Yes', 0: 'No'}
df['decision_label'] = df['dec'].map(decision_mapping)


In [29]:
df_decision_female = df.loc[df['gender'] == "Female"]['decision_label']\
    .value_counts(dropna=False)\
    .reset_index()
df_decision_female['proportion'] = round(100 * df_decision_female['count'] / df_decision_female['count'].sum(), 3)

df_decision_male = df.loc[df['gender'] == "Male"]['decision_label']\
    .value_counts(dropna=False)\
    .reset_index()
df_decision_male['proportion'] = round(100 * df_decision_male['count'] / df_decision_male['count'].sum(), 3)

print("Decision after a speed date, for female")
display(df_decision_female)

print("Decision after a speed date, for male")
display(df_decision_male)


Decision after a speed date, for female


,decision_label,count,proportion
0,No,2655,63.456
1,Yes,1529,36.544


Decision after a speed date, for male


,decision_label,count,proportion
0,No,2205,52.575
1,Yes,1989,47.425


In [30]:
fig = make_subplots(
    rows=1, cols=2, 
    specs=[[{'type':'domain'}, {'type':'domain'}]],  # important pour pie chart
    subplot_titles=["Female", "Male"]
)

fig.add_trace(go.Pie(labels=df_decision_female['decision_label'],
                    values=df_decision_female['proportion'],
                    textinfo='label+percent',
                    marker=dict(colors=['lightgray', 'mediumseagreen'])), 
            row=1, col=1)

fig.add_trace(go.Pie(labels=df_decision_male['decision_label'],
                    values=df_decision_male['proportion'],
                    textinfo='label+percent',
                    marker=dict(colors=['lightgray', 'mediumseagreen'])),
            row=1, col=2)

fig.update_layout(title_text="Decision distribution by genre", title_x=0.5)
fig.show()

The results of the dating experiment show that:
- the overall matching rate is 16.5%
- the mutual rejection rate is twice the matching rate (32%)
- the frustration rate is 51%, meaning that in roughly half of the cases, interest is not reciprocated.
These proportions are consistent for both men and women.

On average, men are more likely to agree to a second date than women: a man's decision is positive in 47% of cases, compared with only 37% for women.

# 4. Detailed analysis

## 4.1. Attributes which matter in a partner

We would like to identify attributes which are the most desirable in a partner and those which are not, and look at whether they differ by gender. 

We will focus on the stated preferences of the participants prior to the experiment, by analyzing the scores of the attributes (`attr1_1`, `sinc1_1`, `intel1_1`, `fun1_1`, `amb1_1`, `shar1_1`).

In [31]:
attributes1 = {'attr1_1' : 'Attractiveness',
        'sinc1_1' : 'Sincerity',
        'intel1_1' : 'Intelligence',
        'fun1_1' : 'Fun',
        'amb1_1' : 'Ambitiousness',
        'shar1_1' : 'Shared interests'
        }

### 4.1.1. Analysis for overall participants

In [32]:
df_sp = df_participant[['iid', 'gender', 'wave', *attributes1.keys()]]
print(df_sp.shape)

(551, 9)


In [33]:
df_sp.describe(include = "all").round(2)

,iid,gender,wave,attr1_1,sinc1_1,intel1_1,fun1_1,amb1_1,shar1_1
count,551.00,551,551.00,544.00,544.00,544.00,543.00,542.00,541.00
unique,NaN,2,NaN,NaN,NaN,NaN,NaN,NaN,NaN
top,NaN,Male,NaN,NaN,NaN,NaN,NaN,NaN,NaN
freq,NaN,277,NaN,NaN,NaN,NaN,NaN,NaN,NaN
mean,276.79,NaN,11.08,22.69,17.29,20.17,17.45,10.81,11.83
std,159.49,NaN,6.01,12.97,7.20,6.88,6.19,6.23,6.47
min,1.00,NaN,1.00,0.00,0.00,0.00,0.00,0.00,0.00
25%,139.50,NaN,6.00,15.00,14.93,17.29,15.00,5.00,8.33
50%,277.00,NaN,11.00,20.00,18.00,20.00,18.00,10.00,11.00
75%,414.50,NaN,15.00,25.00,20.00,23.02,20.00,15.00,16.00


In [34]:
# Displaying observations with missing values in attributes' ratings
df_sp.loc[df_sp[attributes1.keys()].isna().any(axis=1)]

,iid,gender,wave,attr1_1,sinc1_1,intel1_1,fun1_1,amb1_1,shar1_1
312,28,Female,2,NaN,NaN,NaN,NaN,NaN,NaN
828,58,Female,3,NaN,NaN,NaN,NaN,NaN,NaN
838,59,Female,3,NaN,NaN,NaN,NaN,NaN,NaN
1816,129,Male,5,40.0,20.0,20.0,20.0,NaN,NaN
1826,130,Male,5,80.0,10.0,10.0,NaN,NaN,NaN
1866,136,Female,6,NaN,NaN,NaN,NaN,NaN,NaN
5004,339,Male,13,NaN,NaN,NaN,NaN,NaN,NaN
5014,340,Male,13,NaN,NaN,NaN,NaN,NaN,NaN
5114,346,Female,14,NaN,NaN,NaN,NaN,NaN,NaN
7828,528,Female,21,40.0,10.0,30.0,10.0,10.0,NaN


Observations with missing values for all attributes' ratings are excluded.  
For the remaining missing values, we impute zero because the sum of the non missing attributes' ratings is theoretically equal to 100.

In [35]:
# Deleting observations with missing values for all attributes' ratings
df_sp = df_sp.loc[~df_sp[attributes1.keys()].isna().all(axis=1)]
print(df_sp.shape)

# Imputation of the rest of the missing values: they are equal to 0
df_sp[list(attributes1.keys())] = df_sp[list(attributes1.keys())].fillna(0)

(544, 9)


### Checking for the range of the attributes
The descriptive notice indicates that the rating scale differs differs between waves 6 to 9 and the rest of the waves:
- for waves 6 to 9 : the importance of each attribute in a potential date is rated on a scale of 1 to 10
- for the other waves : 100 points are distributed among the six attributes. Total points must equal 100.

We are going to check whether this is the case and deal with it.

In [36]:
# Creating the score (sum of the attributes ratings)
df_sp['score1'] = df_sp['attr1_1'] + df_sp['sinc1_1'] + df_sp['intel1_1'] + df_sp['fun1_1'] + df_sp['amb1_1'] + df_sp['shar1_1']

# Dividing the dataset into two subsets
df_sp_wave_6_9 = df_sp.loc[(df['wave'] >= 6) & (df_sp['wave'] <= 9)]
df_sp_wave_rest = df_sp.loc[~((df['wave'] >= 6) & (df_sp['wave'] <= 9))]

In [37]:
# Displaying the 5 first observations of waves 6-9
df_sp_wave_6_9[[*list(attributes1.keys()), "score1"]].head()

,attr1_1,sinc1_1,intel1_1,fun1_1,amb1_1,shar1_1,score1
1846,16.67,16.67,16.67,16.67,16.67,16.67,100.02
1851,12.77,19.15,17.02,17.02,14.89,19.15,100.00
1856,6.67,20.00,20.00,17.78,20.00,15.56,100.01
1861,18.18,22.73,18.18,13.64,13.64,13.64,100.01
1871,17.39,17.39,15.22,17.39,13.04,19.57,100.00


In [38]:
# Descriptive statistics for wave 6 to 9
display(df_sp_wave_6_9[["score1", *list(attributes1.keys())]].describe().round(2))

,score1,attr1_1,sinc1_1,intel1_1,fun1_1,amb1_1,shar1_1
count,101.00,101.00,101.00,101.00,101.00,101.00,101.00
mean,100.00,16.19,17.97,18.87,17.88,14.73,14.37
std,0.01,3.54,2.74,2.00,2.36,4.13,3.89
min,99.98,6.67,5.13,14.71,12.50,2.33,2.27
25%,100.00,14.55,16.67,17.31,16.67,13.04,12.50
50%,100.00,16.00,17.95,18.75,17.78,15.56,15.09
75%,100.01,18.18,19.57,20.00,19.23,17.78,16.67
max,100.02,27.78,23.81,23.81,27.78,20.59,23.81


In [39]:
# Descriptive statistics for the rest of the waves
display(df_sp_wave_rest[["score1", *list(attributes1.keys())]].describe().round(2))

,score1,attr1_1,sinc1_1,intel1_1,fun1_1,amb1_1,shar1_1
count,443.00,443.00,443.00,443.00,443.00,443.00,443.00
mean,100.12,24.17,17.14,20.47,17.32,9.86,11.17
std,2.86,13.86,7.87,7.53,6.81,6.31,6.84
min,90.00,0.00,0.00,0.00,0.00,0.00,0.00
25%,100.00,15.00,10.00,17.00,12.00,5.00,5.00
50%,100.00,20.00,19.00,20.00,18.00,10.00,10.00
75%,100.00,30.00,20.00,25.00,20.00,15.00,15.00
max,148.00,100.00,60.00,50.00,50.00,53.00,30.00


In [40]:
display(df_sp[["score1", *list(attributes1.keys())]].describe().round(2))
max_threshold = 100
outlier_low = df_sp.loc[df_sp['score1'] < max_threshold]
outlier_high = df_sp.loc[df_sp['score1'] > max_threshold]
print(f"Observations out of the expected value:\n \
    {len(outlier_low)} observations display a total score below {max_threshold}.\n \
    {len(outlier_high)} observations display a total score above {max_threshold}.\n")


,score1,attr1_1,sinc1_1,intel1_1,fun1_1,amb1_1,shar1_1
count,544.00,544.00,544.00,544.00,544.00,544.00,544.00
mean,100.10,22.69,17.29,20.17,17.42,10.77,11.77
std,2.58,12.97,7.20,6.88,6.23,6.25,6.52
min,90.00,0.00,0.00,0.00,0.00,0.00,0.00
25%,100.00,15.00,14.93,17.29,15.00,5.00,8.00
50%,100.00,20.00,18.00,20.00,18.00,10.00,10.76
75%,100.00,25.00,20.00,23.02,20.00,15.00,16.00
max,148.00,100.00,60.00,50.00,50.00,53.00,30.00


Observations out of the expected value:
     30 observations display a total score below 100.
     45 observations display a total score above 100.



For the waves 6 to 9, contrary to what is stated in the dataset description, it seems that the attributes' scores have been rescaled since the sum of the ratings equals to 100.

For the rest of the waves (1-5 and 10-21), we detect observations out of the expected range for individual scores prior to the experiment, i.e. observations with a total score of attributes ratings higher than 100. 

We decide to **rescale the ratings for individual attributes** and **impute the percentage of each attribute in the total score**, so that the sum of the 6 ratings equals 100.


In [41]:
# Rescaling the ratings attributes

# Determining the proportions of each attribute in each total score
df_sp.loc[:, 'attr1_1'] = df_sp['attr1_1'] / df_sp['score1'] * 100
df_sp.loc[:, 'sinc1_1'] = df_sp['sinc1_1'] / df_sp['score1'] * 100
df_sp.loc[:, 'intel1_1'] = df_sp['intel1_1'] / df_sp['score1'] * 100
df_sp.loc[:, 'fun1_1'] = df_sp['fun1_1'] / df_sp['score1'] * 100
df_sp.loc[:, 'amb1_1'] = df_sp['amb1_1'] / df_sp['score1'] * 100
df_sp.loc[:, 'shar1_1'] = df_sp['shar1_1'] / df_sp['score1'] * 100

df_sp.head()

# Removing the old score1 variable
df_sp = df_sp.drop(columns = ['score1'])

# Recreating the total score (sum of the 6 ratings)    
df_sp['score1'] = df_sp['attr1_1'] + df_sp['sinc1_1'] + \
    df_sp['intel1_1'] + df_sp['fun1_1'] + \
    df_sp['amb1_1'] + df_sp['shar1_1']

# Check statistics for the score variable
display(df_sp['score1'].describe().round(2))

count    544.0
mean     100.0
std        0.0
min      100.0
25%      100.0
50%      100.0
75%      100.0
max      100.0
Name: score1, dtype: float64

In [42]:
df_sp_describe = df_sp.describe().round(2).T
df_sp_describe['coeff_of_variation'] = round(df_sp_describe['std']/df_sp_describe['mean'], 2)
df_sp_describe['interquartile_range'] = round(df_sp_describe['75%']-df_sp_describe['25%'], 2)
df_sp_describe = df_sp_describe.T
df_sp_describe

,iid,wave,attr1_1,sinc1_1,intel1_1,fun1_1,amb1_1,shar1_1,score1
count,544.00,544.00,544.00,544.00,544.00,544.00,544.00,544.00,544.0
mean,277.95,11.13,22.67,17.27,20.16,17.41,10.73,11.75,100.0
std,159.42,6.01,12.98,7.19,6.89,6.23,6.08,6.50,0.0
min,1.00,1.00,0.00,0.00,0.00,0.00,0.00,0.00,100.0
25%,141.75,6.75,15.00,14.66,17.24,15.00,5.00,8.00,100.0
50%,277.50,11.00,20.00,18.00,20.00,18.00,10.00,11.06,100.0
75%,416.25,16.00,25.00,20.00,22.80,20.00,15.00,16.00,100.0
max,552.00,21.00,100.00,60.00,50.00,50.00,35.81,30.00,100.0
coeff_of_variation,0.57,0.54,0.57,0.42,0.34,0.36,0.57,0.55,0.0
interquartile_range,274.50,9.25,10.00,5.34,5.56,5.00,10.00,8.00,0.0


Based on the descriptive statistics, we note that :
- attractiveness has the widest range of ratings (from 0 to 100), with a higher mean score and dispersion compared to the other attributes
- attractiveness, ambitiousness and shared interests scores have more variability than intelligence, fun and sincerity.

In [43]:
fig = make_subplots(rows = 6, cols = 1, shared_xaxes = True, vertical_spacing = 0.03)

for i, (var, label) in enumerate(attributes1.items()):
    fig.add_trace(
        go.Box(
            x = df_sp[var],        
            name = label,
            orientation = 'h'   
        ),
        row = i + 1,  
        col = 1
    )

fig.update_layout(
    title = "Attributes scores for overall participants",
    autosize = False,
    width = 750,
    height = 600,
    showlegend = False
)

fig.update_xaxes(
    autorange = True,
    showgrid = True,
    dtick = 10,
    row = 6,  
    col = 1
)
fig.update_layout(title_x=0.5)
fig.show()

In [44]:
# Displaying mean ratings of each attribute
df_attribute_mean = df_sp[list(attributes1.keys())].mean().reset_index().rename(columns = {0: 'score'})
df_attribute_mean['attribute'] = list(attributes1.values())
df_attribute_mean.sort_values(['score'], inplace = True)
display(df_attribute_mean)

fig = px.bar(
    df_attribute_mean,
    x = "score",
    y = "attribute",
    title = "Mean rating of each attribute",
    color_discrete_sequence = ["lightseagreen"]
)
fig.update_layout(title_x=0.5)
fig.show()


,index,score,attribute
4,amb1_1,10.730478,Ambitiousness
5,shar1_1,11.749660,Shared interests
1,sinc1_1,17.273806,Sincerity
3,fun1_1,17.409393,Fun
2,intel1_1,20.162351,Intelligence
0,attr1_1,22.674313,Attractiveness


On average, the ranking of the attributes, from most to less important, is the following : 
- attractiveness
- intelligence
- fun
- sincerity
- shared interests
- ambitiousness.

### 4.1.2. Analysis by gender

Let's see if the ranking of the attributes differ from the gender.

In [45]:
# Descriptive statistics per gender
def describe_attrib_gender(input_df, listvar, gender_value):
    df_temp = input_df.loc[input_df['gender'] == gender_value][[*listvar, "gender"]].drop_duplicates().describe().round(2)
    df_temp_T = df_temp.T
    df_temp_T['coeff_of_variation'] = round(df_temp_T['std']/df_temp_T['mean'], 2)
    df_temp_T['interquartile_range'] = round(df_temp_T['75%']-df_temp_T['25%'], 2)
    return(df_temp_T.T)

print("Descriptive statistics for male")
display(describe_attrib_gender(df_sp, list(attributes1.keys()), "Male"))

print("Descriptive statistics for female")
display(describe_attrib_gender(df_sp, list(attributes1.keys()), "Female"))

Descriptive statistics for male


,attr1_1,sinc1_1,intel1_1,fun1_1,amb1_1,shar1_1
count,253.00,253.00,253.00,253.00,253.00,253.00
mean,27.27,16.24,19.30,17.31,8.66,11.21
std,14.33,7.46,7.10,6.93,5.99,6.85
min,6.67,0.00,0.00,0.00,0.00,0.00
25%,19.00,10.00,15.22,14.71,5.00,5.00
50%,22.22,16.67,20.00,17.39,10.00,10.00
75%,30.00,20.00,22.22,20.00,13.21,16.00
max,100.00,40.00,42.86,50.00,35.81,30.00
coeff_of_variation,0.53,0.46,0.37,0.40,0.69,0.61
interquartile_range,11.00,10.00,7.00,5.29,8.21,11.00


Descriptive statistics for female


,attr1_1,sinc1_1,intel1_1,fun1_1,amb1_1,shar1_1
count,235.00,235.00,235.00,235.00,235.00,235.00
mean,18.17,18.04,21.21,17.47,12.64,12.48
std,10.27,7.37,7.21,5.79,5.56,5.91
min,0.00,0.00,2.00,0.00,0.00,0.00
25%,12.50,15.00,17.57,15.00,10.00,10.00
50%,15.09,18.87,20.00,18.00,15.00,13.00
75%,20.00,20.00,25.00,20.00,16.36,16.00
max,90.00,60.00,50.00,40.00,30.00,30.00
coeff_of_variation,0.57,0.41,0.34,0.33,0.44,0.47
interquartile_range,7.50,5.00,7.43,5.00,6.36,6.00


In [46]:
fig = make_subplots(rows = 6, cols = 1, shared_xaxes = True, vertical_spacing = 0.03)

for i, (var, label) in enumerate(attributes1.items()):
    fig.add_trace(
        go.Box(
            x = df_sp[var],        
            y = df_sp['gender'],   
            name = label,
            orientation = 'h'   
        ),
        row = i + 1,  
        col = 1
    )

fig.update_layout(
    title = "Attributes scores by gender",
    autosize = False,
    width = 750,
    height = 700
)

fig.update_xaxes(
    autorange = True,
    showgrid = True,
    dtick = 10,
    row = 6,  
    col = 1
)
fig.update_layout(title_x=0.5)
fig.show()

In [47]:
# Displaying mean ratings of each attribute, by gender 
df_attribute_mean_by_gender = df_sp.groupby("gender")[list(attributes1.keys())].mean()
df_attribute_mean_by_gender.columns = list(attributes1.values())
df_attribute_mean_by_gender = df_attribute_mean_by_gender.stack().reset_index().rename(columns = {'level_1' : 'attribute', 0: 'rating'})
display(df_attribute_mean_by_gender.head(10).round(2))

fig = px.bar(
    df_attribute_mean_by_gender,
    x = "rating",
    y = "attribute",
    color = "gender",
    barmode = "group",
    text_auto = '.1f',
    title = "Mean rating of each attribute by gender",
    color_discrete_map = {
        "Male": "steelblue",
        "Female": "lightcoral"
    }
)
fig.update_layout(title_x=0.5)
fig.show()

,gender,attribute,rating
0,Female,Attractiveness,18.02
1,Female,Sincerity,18.22
2,Female,Intelligence,20.98
3,Female,Fun,17.30
4,Female,Ambitiousness,12.82
5,Female,Shared interests,12.66
6,Male,Attractiveness,27.23
7,Male,Sincerity,16.34
8,Male,Intelligence,19.37
9,Male,Fun,17.52


On average:

- For men, attractiveness is the most desirable attribute out of the six. They put less importance on ambition and shared interests.

- For women, the scores for each attribute are more evenly spread.
Women put less importance on shared interests and ambition. Intelligence is what they value the most.

- Women value much more ambition than men. Fun has similar importance for both men and women.

## 4.2. Is real behaviour consistent with the stated preferences? 

We focus on revealed preferences of the participants (observed behaviour), by analyzing the correlation between attributes score given during the event (`attr`, `sinc`, `intel`, `fun`, `amb`, `shar`) and the decision of the participant towards his/her date partner (`dec`).

Several analyses can be performed:
- by using box plots, displaying the distribution of the attributes score, depending on the decision of the participant (positive or negative). We search if a particular response tends to be associated with low or high values of the attributes score.
- evaluating, for each attribute, the magnitude of the difference between the mean score associated with a positive decision and the one associated with a negative decision

In [48]:
attributes = {'attr' : 'Attractiveness',
        'sinc' : 'Sincerity',
        'intel' : 'Intelligence',
        'fun' : 'Fun',
        'amb' : 'Ambitiousness',
        'shar' : 'Shared interests'
        }

df_rp = df[['iid', 'wave', 'round', 'order', 'gender', 'dec', *list(attributes.keys())]]
df_rp.shape

(8378, 12)

In [49]:
# Deleting observations with missing values for all attributes' ratings
df_rp = df_rp.loc[~df_rp[attributes.keys()].isna().all(axis=1)]
print(df_rp.shape)

(8186, 12)


In [50]:
# Descriptive statistics for all participants 
df_rp_describe = df_rp.describe().round(2).T
df_rp_describe['coeff_of_variation'] = round(df_rp_describe['std']/df_rp_describe['mean'], 2)
df_rp_describe['interquartile_range'] = round(df_rp_describe['75%'] - df_rp_describe['25%'], 2)
df_rp_describe = df_rp_describe.T
df_rp_describe

,iid,wave,round,order,dec,attr,sinc,intel,fun,amb,shar
count,8186.00,8186.00,8186.00,8186.00,8186.00,8176.00,8101.00,8082.00,8028.00,7666.00,7311.00
mean,281.30,11.26,16.83,8.84,0.43,6.19,7.18,7.37,6.40,6.78,5.47
std,158.19,5.98,4.36,5.43,0.49,1.95,1.74,1.55,1.95,1.79,2.16
min,1.00,1.00,5.00,1.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00
25%,152.00,7.00,14.00,4.00,0.00,5.00,6.00,6.00,5.00,6.00,4.00
50%,278.00,11.00,18.00,8.00,0.00,6.00,7.00,7.00,7.00,7.00,6.00
75%,406.00,15.00,20.00,13.00,1.00,8.00,8.00,8.00,8.00,8.00,7.00
max,552.00,21.00,22.00,22.00,1.00,10.00,10.00,10.00,10.00,10.00,10.00
coeff_of_variation,0.56,0.53,0.26,0.61,1.14,0.32,0.24,0.21,0.30,0.26,0.39
interquartile_range,254.00,8.00,6.00,9.00,1.00,3.00,2.00,2.00,3.00,2.00,3.00


In [51]:
# Descriptive statistics by gender

print("Descriptive statistics for male")
display(describe_attrib_gender(df_rp, list(attributes.keys()), "Male"))

print("Descriptive statistics for female")
display(describe_attrib_gender(df_rp, list(attributes.keys()), "Female"))

Descriptive statistics for male


,attr,sinc,intel,fun,amb,shar
count,3281.00,3241.00,3241.00,3233.00,3086.00,2982.00
mean,6.33,7.19,7.24,6.39,6.52,5.33
std,1.90,1.66,1.51,1.89,1.76,2.12
min,0.00,0.00,0.00,0.00,0.00,0.00
25%,5.00,6.00,6.00,5.00,5.00,4.00
50%,6.00,7.00,7.00,6.00,7.00,5.00
75%,8.00,8.00,8.00,8.00,8.00,7.00
max,10.00,10.00,10.00,10.00,10.00,10.00
coeff_of_variation,0.30,0.23,0.21,0.30,0.27,0.40
interquartile_range,3.00,2.00,2.00,3.00,3.00,3.00


Descriptive statistics for female


,attr,sinc,intel,fun,amb,shar
count,3393.00,3365.00,3347.00,3314.00,3150.00,2973.00
mean,5.77,7.04,7.44,6.16,6.90,5.21
std,2.05,1.92,1.65,2.12,1.91,2.26
min,0.00,0.00,0.00,0.00,0.00,0.00
25%,4.00,6.00,6.00,5.00,6.00,4.00
50%,6.00,7.00,8.00,6.00,7.00,5.00
75%,7.00,8.00,9.00,8.00,8.00,7.00
max,10.00,10.00,10.00,10.00,10.00,10.00
coeff_of_variation,0.36,0.27,0.22,0.34,0.28,0.43
interquartile_range,3.00,2.00,3.00,3.00,2.00,3.00


#### Box plots of the attributes score, depending on the decision of the participant (positive or negative)


In [52]:
fig = go.Figure()

fig = make_subplots(rows = 6, 
                    cols = 1, 
                    subplot_titles = [attributes[attrib] for attrib in list(attributes.keys())])

for i, attribute in enumerate(attributes.keys()):
    row = i + 1
    for dec_value in [0, 1]:
        fig.add_trace(
            go.Box(
                x = df_rp[df_rp['dec'] == dec_value][attribute],
                y = [f'dec={dec_value}'] * df_rp[df_rp['dec'] == dec_value].shape[0],
                orientation = 'h',
                name = f'dec={dec_value}',
                boxmean = True,
                marker_color = 'indianred' if dec_value == 0 else 'seagreen'
            ),
            row = row,
            col = 1
        )
        
fig.update_layout(
    height = 1400,
    width = 700,
    title = "Box plot of attributes score by decision",
    showlegend = False
)
fig.update_layout(title_x=0.5)
fig.show()

A positive decision is likely to be associated with higher attributes' ratings.

Indeed, when the decision is positive (dec = 1), we observe that:
- the range of the attributes is narrower
- the first and the thirds quantiles, the median and the mean are higher

#### Evaluating, for each attribute, the magnitude of the difference between the mean score associated with a positive decision and the one associated with a negative decision

In [53]:
df_rp_groupby_dec = df_rp[['dec', *attributes.keys()]].groupby(['dec']).mean().round(2).reset_index()
df_rp_groupby_dec.columns = ['decision', *attributes.values()]
df_rp_groupby_dec = pd.melt(df_rp_groupby_dec,
                    id_vars = ['decision'],
                    var_name = 'attribute',
                    value_name = 'mean_score'
                    )
df_rp_groupby_dec['decision'] = df_rp_groupby_dec['decision'].map({0: 'Negative', 1: 'Positive'})
display(df_rp_groupby_dec)

fig = px.bar(
    df_rp_groupby_dec,
    x = "attribute",
    y = "mean_score",
    color = "decision",
    barmode = 'group',
    text_auto = True,
    title = "Mean score of each attribute, by decision",
    color_discrete_map = {
        "Negative": "crimson",
        "Positive": "mediumseagreen"
    }
)
fig.update_layout(title_x=0.5)
fig.show()

,decision,attribute,mean_score
0,Negative,Attractiveness,5.37
1,Positive,Attractiveness,7.29
2,Negative,Sincerity,6.86
3,Positive,Sincerity,7.60
4,Negative,Intelligence,7.08
5,Positive,Intelligence,7.76
6,Negative,Fun,5.70
7,Positive,Fun,7.33
8,Negative,Ambitiousness,6.49
9,Positive,Ambitiousness,7.16


In [54]:
df_rp_groupby_dec_pivot = df_rp_groupby_dec.pivot(index = 'attribute', columns = 'decision', values = 'mean_score').reset_index()
df_rp_groupby_dec_pivot['diff_pos_vs_neg'] = df_rp_groupby_dec_pivot['Positive'] - df_rp_groupby_dec_pivot['Negative']
display(df_rp_groupby_dec_pivot.round(2))

fig = px.bar(
    df_rp_groupby_dec_pivot,
    y = "diff_pos_vs_neg",
    x = "attribute",
    barmode = "group",
    text_auto = '.1f',
    color_discrete_sequence = ["lightseagreen"],
    title = "Difference of mean scores for each attribute, between positive and negative response"
)
fig.update_layout(title_x=0.5)
fig.show()


decision,attribute,Negative,Positive,diff_pos_vs_neg
0,Ambitiousness,6.49,7.16,0.67
1,Attractiveness,5.37,7.29,1.92
2,Fun,5.70,7.33,1.63
3,Intelligence,7.08,7.76,0.68
4,Shared interests,4.72,6.46,1.74
5,Sincerity,6.86,7.60,0.74


When looking at the difference of the means scores between positive and negative decisions, we note that:
- it is relatively high for attractiveness, shared interests and fun, so these three attributes are the most differentiating ones in the choice of a partner
- it is lower for ambitiousness, intelligence and sincerity: these attributes can be considered less important.

In [55]:
df_rp_groupby_gender_dec = df_rp[['gender','dec', *attributes.keys()]].groupby(['gender', 'dec']).mean().round(2).reset_index()
df_rp_groupby_gender_dec.columns = ['gender', 'decision', *attributes.values()]
df_rp_groupby_gender_dec = pd.melt(df_rp_groupby_gender_dec,
                    id_vars = ['gender', 'decision'],
                    var_name = 'attribute',
                    value_name = 'mean_score'
                    )
df_rp_groupby_gender_dec['decision'] = df_rp_groupby_gender_dec['decision'].map({0: 'Negative', 1: 'Positive'})
df_rp_groupby_gender_dec_pivot = df_rp_groupby_gender_dec.pivot(index = ['gender','attribute'], columns = 'decision', values = 'mean_score').reset_index()
df_rp_groupby_gender_dec_pivot['diff_pos_vs_neg'] = df_rp_groupby_gender_dec_pivot['Positive'] -df_rp_groupby_gender_dec_pivot['Negative']
display(df_rp_groupby_gender_dec_pivot.round(2))


decision,gender,attribute,Negative,Positive,diff_pos_vs_neg
0,Female,Ambitiousness,6.70,7.37,0.67
1,Female,Attractiveness,5.23,7.07,1.84
2,Female,Fun,5.61,7.38,1.77
3,Female,Intelligence,7.16,7.93,0.77
4,Female,Shared interests,4.70,6.58,1.88
5,Female,Sincerity,6.78,7.63,0.85
6,Male,Ambitiousness,6.24,7.00,0.76
7,Male,Attractiveness,5.53,7.45,1.92
8,Male,Fun,5.80,7.30,1.50
9,Male,Intelligence,6.98,7.62,0.64


In [59]:
fig = px.bar(
    df_rp_groupby_gender_dec_pivot,
    y = "diff_pos_vs_neg",
    x = "attribute",
    color = "gender",
    barmode = "group",
    text_auto = '.1f',
    title = "Difference of mean scores for each attribute, between positive and negative response, by gender",
    color_discrete_map = {
        "Male": "steelblue",
        "Female": "lightcoral"
    }
)
fig.update_layout(title_x=0.5)
fig.show()

Based on revealed preferences, on average, we observe that whatever gender, participants give more importance to attractiveness, shared interests and fun in the choice of a partner, rather than ambitiousness, intelligence and sincerity.

For males, attractiveness is the most desirable attribute, which is consistent with stated preferences regarding this attribute.

We note divergences between participants' behaviour with their stated preferences:
- intelligence seems to be a less desirable attribute in reality
- shared interests seem to be a more desirable attribute in reality

In [58]:
counts = df_rp["attr"].value_counts().sort_index()
counts

attr
0.0        8
1.0      109
2.0      244
3.0      390
3.5        1
4.0      749
5.0     1260
6.0     1658
6.5        7
7.0     1646
7.5        3
8.0     1231
8.5        1
9.0      540
9.5        3
9.9        1
10.0     325
Name: count, dtype: int64

In [56]:
prop_cond = (
    df_rp.groupby("attr")['dec']
      .mean()
)
prop_cond

attr
0.0     0.000000
1.0     0.073394
2.0     0.057377
3.0     0.056410
3.5     0.000000
4.0     0.105474
5.0     0.223016
6.0     0.350422
6.5     0.571429
7.0     0.557108
7.5     1.000000
8.0     0.720552
8.5     1.000000
9.0     0.798148
9.5     0.333333
9.9     1.000000
10.0    0.852308
Name: dec, dtype: float64

## 4.3. Can people accurately predict their own perceived value in the dating market? 

We will compare self rating of attributes (`attr3_1`, `sinc3_1`, `fun3_1`, `intel3_1`, `amb3_1`) with the ratings given by the partners (`attr_o`, `sinc_o`, `fun_o`, `intel_o`, `amb_o`).

We compute and analyze the **difference between the ratings for each attribute**.

In [60]:
cols_to_keep = ['attr3_1', 'sinc3_1', 'fun3_1', 'intel3_1', 'amb3_1',
                'attr_o', 'sinc_o', 'fun_o', 'intel_o', 'amb_o']
df_self_vs_other = df[cols_to_keep]
df_self_vs_other.shape

(8378, 10)

In [61]:
# Deleting observations with missing values for all attributes' ratings
df_self_vs_other = df_self_vs_other.loc[~df_self_vs_other[cols_to_keep].isna().all(axis=1)]
print(df_self_vs_other.shape)

(8370, 10)


In [62]:
for attribute in ['attr', 'sinc', 'fun', 'intel', 'amb']:
    df_self_vs_other[attribute + '_diff'] = df_self_vs_other[attribute + '3_1'] - df_self_vs_other[attribute + '_o']
    
df_self_vs_other.head()

,attr3_1,sinc3_1,fun3_1,intel3_1,amb3_1,attr_o,sinc_o,fun_o,intel_o,amb_o,attr_diff,sinc_diff,fun_diff,intel_diff,amb_diff
0,6.0,8.0,8.0,8.0,7.0,6.0,8.0,8.0,8.0,8.0,0.0,0.0,0.0,0.0,-1.0
1,6.0,8.0,8.0,8.0,7.0,7.0,8.0,7.0,10.0,7.0,-1.0,0.0,1.0,-2.0,0.0
2,6.0,8.0,8.0,8.0,7.0,10.0,10.0,10.0,10.0,10.0,-4.0,-2.0,-2.0,-2.0,-3.0
3,6.0,8.0,8.0,8.0,7.0,7.0,8.0,8.0,9.0,9.0,-1.0,0.0,0.0,-1.0,-2.0
4,6.0,8.0,8.0,8.0,7.0,8.0,7.0,6.0,9.0,9.0,-2.0,1.0,2.0,-1.0,-2.0


In [63]:
df_self_vs_other_describe = df_self_vs_other[[
    'attr_diff', 
    'sinc_diff', 
    'fun_diff', 
    'intel_diff', 
    'amb_diff' 
    ]].describe().round(3)

In [64]:
df_self_vs_other_describe

,attr_diff,sinc_diff,fun_diff,intel_diff,amb_diff
count,8069.000,7997.000,7923.000,7978.000,7563.000
mean,0.902,1.126,1.309,1.037,0.810
std,2.192,2.238,2.320,1.869,2.461
min,-7.000,-8.000,-7.000,-6.000,-8.000
25%,-1.000,0.000,0.000,0.000,-1.000
50%,1.000,1.000,1.000,1.000,1.000
75%,2.000,3.000,3.000,2.000,2.000
max,9.000,10.000,9.000,10.000,9.000


In [65]:
labels = {
    'attr': 'Attractiveness',
    'sinc': 'Sincerity',
    'fun': 'Fun',
    'intel': 'Intelligence',
    'amb': 'Ambitiousness'
}

fig = go.Figure()

for i, attribute in enumerate(labels.keys()):
    fig.add_trace(go.Box(
        y = df_self_vs_other[attribute + '_diff'],
        name = labels[attribute],
        boxmean = True # mean is represented in dashed lines
))

fig.update_layout(
    title = "Differences between self perception and others' perception",
    title_x=0.5,
    yaxis_title = "Difference score",
    showlegend=False
)
fig.show()

On average, participants are fairly confident and tend to overestimate their attributes.
For all five attributes, the average difference rating between self perceived and others' evaluation is around 1 point out of 10.

## 4.4. Impact of the round order on the decision to ask for a second date

We would like to know if the round order of the date influences the decision to respond positively: is it better to be someone's first speed date of the night or their last?

### 4.4.1. Is it better of be someone's first speed date of the night or their last?

In [66]:
cols_to_keep = ['order', 'round', 'dec']
df_first_last = df[cols_to_keep]

In [67]:
df_first_last = df_first_last.loc[(df_rp['order'] == 1) | (df_first_last['order'] == df_first_last['round'])]
df_first_last.loc[df_first_last['order'] == 1, 'order_lib'] = "First"
df_first_last.loc[df_first_last['order'] == df_first_last['round'], 'order_lib'] = "Last"
display(df_first_last.head())
print("Shape of the subset:", df_first_last.shape)


,order,round,dec,order_lib
2,10,10,1,Last
6,1,10,1,First
10,10,10,0,Last
13,1,10,1,First
28,10,10,0,Last


Shape of the subset: (1062, 4)


In [68]:
for lib in ['First', 'Last']:
    print("Descriptive statistics of decision for", lib.lower(), "order")
    display(df_first_last.loc[df_first_last['order_lib'] == lib]['dec'].describe().round(3))

Descriptive statistics of decision for first order


count    548.000
mean       0.502
std        0.500
min        0.000
25%        0.000
50%        1.000
75%        1.000
max        1.000
Name: dec, dtype: float64

Descriptive statistics of decision for last order


count    514.000
mean       0.463
std        0.499
min        0.000
25%        0.000
50%        0.000
75%        1.000
max        1.000
Name: dec, dtype: float64

In [69]:
# Proportion of positive answer per order
df_first_last_answer = df_first_last.groupby(['order_lib'], observed = True, dropna = False).agg(
    nb_positive_answer = ("dec", "sum"),
    nb_answer = ("dec", "count")
).reset_index()
df_first_last_answer['positive_answer_rate'] = round(df_first_last_answer['nb_positive_answer']/df_first_last_answer['nb_answer'] * 100, 1)
display(df_first_last_answer)


fig = px.bar(
    df_first_last_answer,
    x = "order_lib",
    y = "positive_answer_rate",
    text_auto = '.1f',
    title = "Proportion of positive answer depending on the order (first or last)",
    color_discrete_sequence = ["lightseagreen"]
)

fig.update_layout(
    title_x=0.5,
    xaxis_title = "Order",
    yaxis_title = "Positive answer rate (%)",
)
fig.show()


,order_lib,nb_positive_answer,nb_answer,positive_answer_rate
0,First,275,548,50.2
1,Last,238,514,46.3


Being on first position seems to be more favourable to get a second date: on average, the participant is likely to request a second date half of the time.
Being on last position is associated with a lower proportion of positive answer. One explanation could be social tiredness.

### 4.4.2. Does the order in the round influence the decision to respond positively?

In [70]:
# Number of people met during each wave
df_round_wave = df_rp[['wave', 'round']].groupby('wave', dropna = False).agg(
    min_number_people_met = ("round", "min"),
    max_number_people_met = ("round", "max"),
).reset_index()
display(df_round_wave)

# Barplot of the minimum number of people met, per wave
fig = px.bar(df_round_wave, 
            x = 'wave', 
            y = 'min_number_people_met',
            title = "Minimum number of people met per wave")
fig.update_layout(title_x=0.5)
fig.show()


,wave,min_number_people_met,max_number_people_met
0,1,10,10
1,2,16,19
2,3,10,10
3,4,18,18
4,5,10,10
5,6,5,5
6,7,16,16
7,8,10,10
8,9,20,20
9,10,9,9


In each wave, the number of people met per participant is variable, so we cannot directly work with the `order` variable. 

Thus we compute for each participant the **progression in the round**, by dividing the order by the number of people met during the round. This variable varies from 0 to 1: a value near to 0 means the participant is at the beginning of the round whereas a value near to 1 means that he/she is towards the end.

In [71]:
# Create a variable of progression during the experiment
df_rp.loc[:, 'round_progression'] = df_rp['order'] / df_rp['round']

In [72]:
# Defining class of progression order

labels = ["[0 - 0.1[",
        "[0.1 - 0.2[",
        "[0.2 - 0.3[",
        "[0.3 - 0.4[",
        "[0.4 - 0.5[",
        "[0.5 - 0.6[",
        "[0.6 - 0.7[",
        "[0.7 - 0.8[",
        "[0.8 - 0.9[",
        ">= 0.9"
        ]

df_rp['progression_round_class'] = pd.cut(
    x = df_rp['round_progression'],
    bins = np.arange(0, 1.1, 0.1),
    labels = labels
)

In [73]:
df_rp.head()

,iid,wave,round,order,gender,dec,attr,sinc,intel,fun,amb,shar,round_progression,progression_round_class
0,1,1,10,4,Female,1,6.0,9.0,7.0,7.0,6.0,5.0,0.4,[0.3 - 0.4[
1,1,1,10,3,Female,1,7.0,8.0,7.0,8.0,5.0,6.0,0.3,[0.2 - 0.3[
2,1,1,10,10,Female,1,5.0,8.0,9.0,8.0,5.0,7.0,1.0,>= 0.9
3,1,1,10,5,Female,1,7.0,6.0,8.0,7.0,6.0,8.0,0.5,[0.4 - 0.5[
4,1,1,10,7,Female,1,5.0,6.0,7.0,7.0,6.0,6.0,0.7,[0.6 - 0.7[


In [74]:
# Displaying the number of observations per class
df_rp['progression_round_class'].value_counts(dropna = False)

progression_round_class
>= 0.9         964
[0.4 - 0.5[    904
[0.1 - 0.2[    877
[0.3 - 0.4[    865
[0.7 - 0.8[    819
[0.8 - 0.9[    818
[0.6 - 0.7[    815
[0.2 - 0.3[    797
[0.5 - 0.6[    717
[0 - 0.1[      610
Name: count, dtype: int64

In [75]:
# Proportion of positive answer per class of progression
df_order_answer = df_rp.groupby(['progression_round_class'], observed = True, dropna = False).agg(
    nb_positive_answer = ("dec", "sum"),
    nb_answer = ("dec", "count")
).reset_index()
df_order_answer['positive_answer_rate'] = round(df_order_answer['nb_positive_answer']/df_order_answer['nb_answer'] * 100, 1)
df_order_answer

,progression_round_class,nb_positive_answer,nb_answer,positive_answer_rate
0,[0 - 0.1[,292,610,47.9
1,[0.1 - 0.2[,367,877,41.8
2,[0.2 - 0.3[,331,797,41.5
3,[0.3 - 0.4[,371,865,42.9
4,[0.4 - 0.5[,398,904,44.0
5,[0.5 - 0.6[,313,717,43.7
6,[0.6 - 0.7[,338,815,41.5
7,[0.7 - 0.8[,333,819,40.7
8,[0.8 - 0.9[,327,818,40.0
9,>= 0.9,442,964,45.9


In [76]:
fig = px.bar(
    df_order_answer,
    x = "progression_round_class",
    y = "positive_answer_rate",
    title = "Proportion of positive answer depending on the progression round",
    text_auto = '.1f',
    color_discrete_sequence = ["lightseagreen"]
)

fig.update_layout(
    title_x=0.5,
    xaxis_title = 'Round progression',
    yaxis_title = 'Percentage of positive answer (%)'
)

fig.show()

Being either at the beginning or at the end of the night seems to be better positions that being in the middle. Indeed, the positive answer rates are slightly lower in an intermediate position.

# 5. Conclusion

**Stated preferences concerning attributes in a partner:**  
- Overall ranking: attractiveness, intelligence, fun, sincerity, shared interests, ambitiousness.  
- Men favour attractiveness, women value intelligence and ambition more.

**Revealed preferences (actual choices):**  
- Overall, both genders prioritize attractiveness, fun, shared interests when choosing partners. Ambition, intelligence, and sincerity are less influential.  
- For males, attractiveness is the most desirable attribute, which is consistent with stated preferences regarding this attribute.  
- Divergences: intelligence is less important in reality than stated, and shared interests are more important in reality.  

**Self-perception:** participants slightly overestimate their own attributes (the average difference rating between self perceived and others' evaluation is around 1 point on a 10-point scale).

**Round order impact:** being first or last in the speed dating sequence seems to increase chances of a positive second date; middle positions are slightly less favorable.
